core/src/test/scala/kanaloa/reactive/dispatcher/DispatcherSpec.scala

```diff
@@ -192,6 +192,66 @@ class DispatcherSpec extends SpecWithActorSystem with OptionValues {
 
       (received.length.toDouble / numOfWork.toDouble) shouldBe 0.5 +- 0.07
     }
+
+    //todo: move this to integration test once the integration re-org test PR is merged.
+    "start to reject work when worker creation fails" in new ScopeWithActor with Eventually {
+      val failingBackend = new Backend {
+        def apply(f: ActorRefFactory): Future[ActorRef] = Future.failed(new Exception("failing backend"))
+      }
+      val dispatcher = system.actorOf(PushingDispatcher.props(
+        "test",
+        failingBackend,
+        ConfigFactory.parseString(
+          """
+            |kanaloa.default-dispatcher {
+            |  updateInterval = 10ms
+            |  backPressure {
+            |    durationOfBurstAllowed = 10ms
+            |    referenceDelay = 2s
+            |  }
+            |}""".stripMargin
+        )
+      )(ResultChecker.complacent))
+
+      eventually {
+        (1 to 100).foreach(_ ⇒ dispatcher ! "a work")
+        expectMsgType[WorkRejected](20.milliseconds)
+      }(PatienceConfig(5.seconds, 40.milliseconds))
+
+    }
+
+    //todo: move this to integration test once the integration re-org test PR is merged. 
+    "be able to pick up work after worker finally becomes available" in new ScopeWithActor with Eventually with Backends {
+      import scala.concurrent.ExecutionContext.Implicits.global
+      val dispatcher = system.actorOf(PushingDispatcher.props(
+        "test",
+        delayedBackend(500),
+        ConfigFactory.parseString(
+          """
+            |kanaloa.default-dispatcher {
+            |  updateInterval = 50ms
+            |  backPressure {
+            |    durationOfBurstAllowed = 30ms
+            |    referenceDelay = 1s
+            |  }
+            |}""".stripMargin
+        )
+      )(ResultChecker.complacent))
+
+      //reach the point that it starts to reject work
```

Comment:
One thing to think about is to maybe change the implementation of the `Backends` class so that you can control exactly when Backends are created.  You could do something like store Promises in the `Backends` class, and then, when you want to actually create the Backends, invoke a function which will then go through and fill all the promises.  Pretty simple to implement and allows for very fine grained control. 

This same concept can then be expanded for a Backend which fails all creations, until you flip a flag on it, and then it will successfully create Backends.  Again, this is for more fine grained control.